codec/src/main/java/io/netty/handler/codec/string/StringEncoder.java

```diff
@@ -68,7 +69,7 @@ public StringEncoder(Charset charset) {
         if (charset == null) {
             throw new NullPointerException("charset");
         }
-        this.charset = charset;
+        this.encoder = CharsetUtil.encoder(charset);
```

Comment:
@johnou this is not thread-safe... You will need todo this in the `handlerAdded(...)` method as `CharsetUtil.encoder(...)` looks up its encoder via a thread-local. So we need to ensure we call this from the same thread as we use to encode later on.

Reply:
@normanmaurer ptal

Reply:
Why not look this up each time?  Can't this handler be added to multiple event loops?

Reply:
@johnou @carl-mastrangelo is right... it is `@Sharable` so this will not work.

Reply:
@normanmaurer so to avoid breaking changes removing the TODO probably makes most sense then?